modules/tinymce/src/core/main/ts/api/dom/DOMUtils.ts

```diff
@@ -1052,38 +1058,54 @@ function DOMUtils(doc: Document, settings: Partial<DOMUtilsSettings> = {}): DOMU
 
   const createRng = () => doc.createRange();
 
-  const split = (parentElm: Node, splitElm: Node, replacementElm?: Node) => {
-    let r = createRng(), bef, aft, pa;
+  const split = <T extends Node>(parentElm: Node, splitElement: T, replacementElement?: T, optionsInput?: SplitOptions): T => {
```

Comment:
This new logic/option should have explicit tests added (eg in `DOMUtilsTest` or similar), so tests please :smiley:

Reply:
But it is tested by way of the formatremover, isn't it?

Reply:
That only partly tests it and doesn't test it in isolation. This is a public API so it needs to be tested by itself without other code influencing it.

Reply:
Fair enough. Will do.